main/lsp/LSPTypecheckerCoordinator.cc

```diff
@@ -1,22 +1,96 @@
 #include "main/lsp/LSPTypecheckerCoordinator.h"
+#include "main/lsp/LSPTask.h"
 
 #include "absl/synchronization/notification.h"
 
 namespace sorbet::realmain::lsp {
 using namespace std;
 
 namespace {
-// TODO(jvilk): Switch LSPTypecheckerCoordinator interface to use a type of task rather than lambdas.
-class LambdaTask : public core::lsp::Task {
-private:
-    const function<void()> lambda;
+class TypecheckerTask final : public core::lsp::Task {
```

Comment:
Wraps an `LSPTask`, which accepts a `LSPTypecheckerDelegate`, in the generic `Task` interface used elsewhere.